### PR TITLE
Remaster Auto Config

### PR DESCRIPTION
**Pull request by psxcode**
- Add ability to override `initialType`, `zeroBreakingChangeType` and `shouldAlwaysBumpDependents` for each package.
- Add ability to override `npm` config for each package.
- Add `initial` field to Slack colors.
- Change `auto` config to be independent of Start config.
- Change `zeroBreakingChange` to bump package version exactly as defined.
- Remove configurable `autoNamePrefix`, since it was never used.
- Remove `gitOptions`, `workspaceOptions`, `bumpOptions`.
- Remove support for `publishConfig` in favor of `npm` config override.
- Fix issues with `initial` bump type.

### CODE REVIEW

File: packages/auto/npm/src/publish-package.ts [hunk 1]
```diff
@@ -12,20 +13,41 @@ export const publishPackage = async (bumpPackage: TPackageBump, npmOptions?: TNp
...
 
-  if (npmOptions && npmOptions.registry) {
-    options.registry = npmOptions.registry
+    if (isString(npmOptions.publishSubDirectory)) {
+      options.publishSubDirectory = npmOptions.publishSubDirectory
+    }
+
+    if (isString(npmOptions.access)) {
+      options.access = npmOptions.access
+    }
```
Comment (deepsweet) [inline packages/auto/npm/src/publish-package.ts]:
```ts
  const options = {
    registry: 'https://registry.npmjs.org/',
    publishSubDirectory: '',
    access: 'restricted',
  }

  // overrides for root auto config
  if (isDefined(npmOptions)) {
    if (isString(npmOptions.registry)) {
      options.registry = npmOptions.registry
    }

    if (isString(npmOptions.publishSubDirectory)) {
      options.publishSubDirectory = npmOptions.publishSubDirectory
    }

    if (isString(npmOptions.access)) {
      options.access = npmOptions.access
    }
  }
```
⬇️ 

```ts
  const options = {
    registry: 'https://registry.npmjs.org/',
    publishSubDirectory: '',
    access: 'restricted',
    ...npmOptions
  }
```

❓

File: packages/auto/utils/src/compile-auto-config.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  initialType: 'minor',
+  zeroBreakingChangeType: 'minor',
+}
+
+export const compileAutoConfig = (rootPackageConfig?: TAutoConfig, packagePackageConfig?: TAutoConfig): TAutoConfig => {
+  return {
+    ...defaultAutoConfig,
+    ...(isDefined(rootPackageConfig) ? rootPackageConfig : {}),
+    ...(isDefined(packagePackageConfig) ? packagePackageConfig : {}),
+  }
```
Comment (deepsweet) [inline packages/auto/utils/src/compile-auto-config.ts]:
```ts
  return {
    ...defaultAutoConfig,
    ...(isDefined(rootPackageConfig) ? rootPackageConfig : {}),
    ...(isDefined(packagePackageConfig) ? packagePackageConfig : {}),
  }
```

⬇️ 

```ts
  return {
    ...defaultAutoConfig,
    ...{ ...rootPackageConfig, ...packagePackageConfig },
  }

❓ 
```

File: packages/auto/npm/src/publish-package.ts (hunk 1 above)
Comment (psxcode) [reply packages/auto/npm/src/publish-package.ts]: Resolved

File: packages/auto/utils/src/compile-auto-config.ts (hunk 2 above)
Comment (psxcode) [reply packages/auto/utils/src/compile-auto-config.ts]: Resolved

**Pull request closed without merging**